library/core/src/ptr/non_null.rs

```diff
@@ -1416,6 +1418,8 @@ impl<T: ?Sized> NonNull<T> {
     #[must_use]
     #[unstable(feature = "pointer_is_aligned_to", issue = "96284")]
     #[rustc_const_unstable(feature = "const_pointer_is_aligned", issue = "104203")]
+    #[kani::requires(align.is_power_of_two() && align > 0)] // Ensure alignment is a power of two and not zero
```

Comment:
For is `is_power_of_two()` to be true, alignment has to be greater than 0.

Reply:
you mean like changing to requires(align > 0 && align.is_power_of_two())]?

Reply:
no, I mean that you don't need to check that `align > 0` because `align.is_power_of_two()` will already return false for `0`.